### PR TITLE
Storing fileId in FileVersion

### PR DESCRIPTION
**Pull request by iisuslik43**

### CODE REVIEW

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt [hunk 1]
```diff
@@ -603,10 +604,12 @@ class CosmasGoogleCloudService(
             latestVersion.patchesList.last().userName
         }
 
+        val latestFileId = if (latestVersion.fileId.isBlank()) fileId else latestVersion.fileId
```
Comment (dbarashev) [inline src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt:614]:
```suggestion
        // In most cases latestVersion.fileId == fileId
        // However, when file changes its id (e.g. because of Dropbox import), we can
        // read its latest blob thanks to fileIdChangeMap, and latestVersion from this blob
        // will have different fileId, so we should add to latestVersion's window FileVersionInfo
        // with fileId of latestVersion.
        // In reality this makes the latest version unavailable, while previous versions
        // are still available.
        val latestFileId = if (latestVersion.fileId.isBlank()) fileId else latestVersion.fileId
```

**Review by dbarashev — APPROVED** [review]
I added a comment explaining the logic with latestFileId. Please check if my understanding is correct. Also, it seems that we have no unit test covering this edge case, do we? Changes in the test code look pretty technical.

File: src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt (hunk 1 above)
Comment (iisuslik43) [reply src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt:614]: I changed your comment a bit. About testing, `hardRestoring` checks that fileVersionList is correct for this case
